src/main/scala/com/typesafe/sbt/packager/archetypes/JavaServerApplication.scala

```diff
@@ -59,27 +57,30 @@ object JavaServerAppPackaging extends AutoPlugin {
     },
     // === etc config mapping ===
     bashScriptEnvConfigLocation := Some("/etc/default/" + (packageName in Linux).value),
-    linuxEtcDefaultTemplate <<= sourceDirectory map { dir =>
-      val overrideScript = dir / "templates" / ETC_DEFAULT
-      if (overrideScript.exists) overrideScript.toURI.toURL
-      else etcDefaultTemplateSource
-    },
-    linuxStartScriptName := None,
-    makeEtcDefault <<= (packageName in Linux, target in Universal, linuxEtcDefaultTemplate, linuxScriptReplacements)
-      map makeEtcDefaultScript,
-    linuxPackageMappings <++= (makeEtcDefault, bashScriptEnvConfigLocation) map { (conf, envLocation) =>
-      val mapping = for (
-        path <- envLocation;
-        c <- conf
-      ) yield LinuxPackageMapping(Seq(c -> path), LinuxFileMetaData(Users.Root, Users.Root, "644")).withConfig()
-
-      mapping.toSeq
-    }
 
+    linuxStartScriptName := None
+  )
+
+  /* etcDefaultConfig is dependent on serverLoading (systemd, systemv, etc.),
+   * and is therefore distro specific. As such, these settings cannot be defined
+   * in the global config scope. */
+  private val etcDefaultConfig: Seq[Setting[_]] = Seq(
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [List(etcDefaultConfig) should have a smaller scope by using private[this]](https://www.codacy.com/app/nepomukseiler/sbt-native-packager/file/2094166179/issues/source?bid=2040925&fileBranchId=3140198#l67)